### DISCUSSION TITLE
New artifact associated with secondary skill Eagle Eye

### DISCUSSION THREAD

**Discussion by ghost** [ihhub/fheroes2#7464] [Ideas]
I wanted to ask if the new artifact is acceptable from the point of view of developers ? The essence of the artifact - in the presence of a secondary skill Eagle Eye gives resistance to spells of the appropriate level and with the appropriate probability.

**Comment by Branikolog** [comment]
Hi, @Ben80-hero !
I'm not against adding such artifact, but why not just add an artifact, that gives the resistance on its own nature? Or separate resistance skill? Adding additional rules to existing mechanics or skill behaviour, depending on a presence of an artifact, looks quite a questionable, regarding game design.

**Comment by ghost** [comment]
> I'm not against adding such artifact, but why not just add an artifact, that gives the resistance on its own nature? Or separate resistance skill? Adding additional rules to existing mechanics or skill behaviour, depending on a presence of an artifact, looks quite a questionable, regarding game design.

@Branikolog,
The whole point of the idea is to improve the Eagle Eye skill, but since it’s forbidden to do it directly, here’s an option ... Similarly, for example, one could add an artifact that doubles mana income (here, interaction with the Mysticism skill).
In the original game, by the way, often artifacts strengthen spells, and this is quite interesting.

**Reply by Branikolog** [reply]
The majority of artifacts improving the secondary skills affect the skill native behaviour. Improve the % of the skill, for instance. I can hardly remember any artifact, that adds completely new features to existing skills. 
The same I can tell about spells: artifact adds damage to the particular spell or makes cheaper cast for some group of spells. But there're no artifacts, that changes the behaviour of a spell. Like Slow would decrease target defense. Such approach is not really good, from a game design point, since player should consider the possibility of facing some additional effects for the well known spell.

**Reply by Mr-Bajs** [reply]
Also adding secondary effects for spells and such would be rather hard to design the AI to utilise in a good way?

**Comment by ghost** [comment]
> The majority of artifacts improving the secondary skills affect the skill native behaviour. Improve the % of the skill, for instance. I can hardly remember any artifact, that adds completely new features to existing skills. The same I can tell about spells: artifact adds damage to the particular spell or makes cheaper cast for some group of spells. But there're no artifacts, that changes the behaviour of a spell.

Yes, it is true, but to engage in such self-restraint is the path to "defeat". I also tried to limit myself all the time when writing plugins for Heroes 3, until I finally realized that there is no "modding without modding", there is no "true modding" - there is just modding, and creating an addon is modding.

>Such approach is not really good, from a game design point, since player should consider the possibility of facing some additional effects for the well known spell.

In terms of game design, the original game has much more serious problems ;)

**Reply by ihhub** [reply]
@Ben80-hero , please do not mix modding (modification of an existing object) and expansion (adding new content on top of existing). fheroes2 is not a modding project but a game engine recreation. Adding new content is an expansion and for this matter we are going to add something new to compensate some flaws in balance if any.

**Comment by ghost** [comment]
> @Ben80-hero , please do not mix modding (modification of an existing object) and expansion (adding new content on top of existing). fheroes2 is not a modding project but a game engine recreation. Adding new content is an expansion and for this matter we are going to add something new to compensate some flaws in balance if any.

@ihhub , I agree, but everything that I said about unnecessary self-restrictions (for example, in the sense of the features of artifacts), I can say also in relation to expansion.

**Comment by Mr-Bajs** [comment]
I think from a player perspective effects should be simple and easy to understand. Artifacts thats requires the hero to have an secondary skill to utilise the artifact effects should be avoided. An reasonable exception from that could be an artifact that improves the heroes necromancy skull, i my mind.

**Reply by ghost** [reply]
@Branikolog  and @Mr-Bajs ,

maybe you are right, maybe not. Thanks, I was interested to know your opinion.

**Reply by zenseii** [reply]
There kind of are artifacts that require secondary skills. I'm thinking about those that you can only be picked up by a "true leader" etc. Further development of this thing could be discussed.

**Comment by ghost** [comment]
> Also adding secondary effects for spells and such would be rather hard to design the AI to utilise in a good way?

here for the AI ​​it is enough just the conditions when choosing SS - if the hero (or other heroes of this color) has such an artifact - then the value of the EE skill increases.
